nio/rooms.py

```diff
@@ -204,30 +205,34 @@ def is_group(self):
         """
         return not self.is_named
 
-    def add_member(self, user_id, display_name, avatar_url):
-        # type (str, str, str) -> bool
+    def add_member(self, user_id, display_name, avatar_url, invited=False):
+        # type (str, str, str, bool) -> bool
         if user_id in self.users:
             return False
 
         level = self.power_levels.users.get(
             user_id,
-            self.power_levels.defaults.users_default
+            self.power_levels.defaults.users_default,
         )
 
-        user = MatrixUser(user_id, display_name, avatar_url, level)
+        user = MatrixUser(user_id, display_name, avatar_url, level, invited)
         self.users[user_id] = user
 
+        if invited:
+            self.invited_users[user_id] = user
```

Comment:
This will put the user in both the normal member list as well as in the `invited_users` member list. Any reasons why to keep them in both instead of just one of them? Looks good otherwise.

Reply:
Mostly for convenience. I thought it would be good for consistency too, since clients have both `rooms` and `invited_rooms` attributes.